common:
    lidar_topic:  "/points_raw"
    imu_topic:  "/imu_raw"
    map_frame:  "camera_init"
    body_frame:  "body"
    timedelay_lidar2imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).

preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 32
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    timestamp_unit: 2            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 2
    det_range: 100.0

mapping:
    point_skip_num: 4
    surf_frame_ds_res: 0.5
    max_iteration: 3
    ikdtree_resolution: 0.5
    lidar_model_search_range: 5
    cube_side_length: 1000
    runtime_log_enable: 0

    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [ 0, 0, 0.28]
    extrinsic_R: [ 1, 0, 0, 
                   0, 1, 0, 
                   0, 0, 1]

localization:
    gravity: [ 0.0425684, 0.116608, -9.80821 ]

scan_context:
    lidar_height: 2
    sc_dist_thres: 0.5

publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.

official:
    save_globalmap_en: false
    save_resolution: 0.1
    save_keyframe_en: false
